events/app_watcher.go

```diff
@@ -130,4 +178,8 @@ func (m *AppWatcher) scaleTo(newInstanceCount int) {
 
 func (m *AppWatcher) unregisterInstanceMetrics(instanceIndex int) {
 	m.registerer.Unregister(m.MetricsForInstance[instanceIndex].Cpu)
+	m.registerer.Unregister(m.MetricsForInstance[instanceIndex].DiskBytes)
+	m.registerer.Unregister(m.MetricsForInstance[instanceIndex].DiskUtilization)
+	m.registerer.Unregister(m.MetricsForInstance[instanceIndex].MemoryBytes)
+	m.registerer.Unregister(m.MetricsForInstance[instanceIndex].MemoryUtilization)
```

Comment:
It feels weird that NewInstanceMetrics registers the metrics but AppWatcher unregisters them. I think the behaviour should live all in one place - either AppWatcher or InstanceMetrics. (I know it was like that already)